src/rulesets/Base/base.jl

```diff
@@ -88,3 +88,55 @@ function rrule(::typeof(identity), x::Tuple)
     end
     return x, identity_pullback
 end
+
+#####
+##### `evalpoly`
+#####
+
+if VERSION ≥ v"1.4"
+    function frule((_, Δx, Δp), ::typeof(evalpoly), x, p)
+        q = _evalpoly_dxcoef(p)
+        return evalpoly(x, p), evalpoly(x, Δp) + evalpoly(x, q) * Δx
+    end
+
+    function rrule(::typeof(evalpoly), x, p)
+        function evalpoly_pullback(Δy)
+            ∂x = Thunk() do
+                q = _evalpoly_dxcoef(p)
+                return evalpoly(x, q)' * Δy
+            end
+            ∂p = @thunk _evalpoly_backp(Δy, x, p)
+            return NO_FIELDS, ∂x, ∂p
+        end
+        return evalpoly(x, p), evalpoly_pullback
+    end
+
+    _evalpoly_dxcoef(p) = ntuple(i -> i * p[i + 1], length(p) - 1)
+    function _evalpoly_dxcoef(p::AbstractVector)
+        N = length(p)
+        @inbounds q = (1:(N - 1)) .* view(p, 1:(N - 1))
+        return q
+    end
+
+    # This is a geometric progression, that is ∂p = Δy * (I, x, x², …, xⁿ)'
+    # TODO: Handle case where x is a matrix and p is UniformScaling
+    function _evalpoly_backp(Δy, x, p::Tuple)
+        N = length(p)
+        ∂p = ntuple(_ -> Δy, N)
+        x′ = x'
+        for i in 2:N
+            ∂p = Base.setindex(∂p, ∂p[i - 1] * x′, i)
+        end
+        return ∂p
```

Comment:
Pullbacks can't return `Tuple` as a `Tuple` is not a valid differential type,
because `Tuple` do not define `+` or `zero`
Instread one should return a `Composite`

```suggestion
        return Composite{typeof(p)}(∂p...)
```
or
```suggestion
        return Composite{typeof(p), typeof(∂p)}(∂p)
```

Reply:
> Pullbacks can't return `Tuple` as a `Tuple` is not a valid differential type,
> because `Tuple` do not define `+` or `zero`
> Instread one should return a `Composite`

Ah good point, that's what I was missing.

>(in contrast, our, or at least my, opinion on the handling of complex numbers is much less strong. But does still disagree with Zygote.)

Can I read about those disagreements somewhere?

Reply:
Does the input differential for `p` in the `frule` likewise need to be a `Composite` when `p` is a tuple?

Reply:
It should be be a `Composite` yes, but it is less clear to me if it should be constrained to be.